src/js/components/awardv2/idv/activity/ActivityChartTooltip.jsx

```diff
@@ -144,21 +192,41 @@ export default class IdvActivityTooltip extends React.Component {
             xPosition -= overshowing;
         }
         // if x position of off the page, move it on the page
+        // '-' means negative and we are off the page
         if (xPosition.toString()[0] === '-') {
             const difference = 0 - xPosition;
             // adjust the tooltip width based on x position movement
             theTooltipWidth += difference;
             xPosition = 0;
         }
-
-        const startDateDiv = this.startDateDiv.getBoundingClientRect().width;
-        const endDateDiv = this.endDateDiv.getBoundingClientRect().width;
-        const amountsDiv = this.amountsDiv.getBoundingClientRect().width;
         // truncate text when the tooltip is bigger than the graph
         if (data.graphWidth < theTooltipWidth) {
-            await this.truncateText(awardingAgencyNameDiv, recipientNameDiv, theTooltipWidth);
+            // getting the last row, only need if we truncate
+            const startDateDiv = this.startDateDiv.getBoundingClientRect().width;
+            const endDateDiv = this.endDateDiv.getBoundingClientRect().width;
+            const amountsDiv = this.amountsDiv.getBoundingClientRect().width;
+            // getting the first row, only need to if we truncate
+            const piidDiv = this.piidDiv.getBoundingClientRect().width;
+            // const grandparentDiv = this.grandparentDiv.getBoundingClientRect().width;
+            const parentDiv = this.parentDiv.getBoundingClientRect().width;
+            // this is the width of the last row
+            // the smallest width allowed since we have to keep it inline
+            // 100px = 10px ( Left of Row ) + 90px ( Three divs padding right )
+            theTooltipWidth = startDateDiv + endDateDiv + amountsDiv + 100;
+            // array of divs to truncate
+            let truncateTheseDivWidths = [
+                awardingAgencyNameDiv,
+                recipientNameDiv,
+                piidDiv,
+                parentDiv
+            ];
+            // do not truncate PIID if there is no grandparent
+            if (!this.grandparentDiv) {
+                truncateTheseDivWidths = [awardingAgencyNameDiv, recipientNameDiv];
```

Comment:
Shouldn't this be `[recipientNameDiv, awardingAgencyNameDiv]`? Since we're relying on the index value of `arrayOfProperties` in line 124?